src/ChatGptNet/ChatGptServiceCollectionExtensions.cs

```diff
@@ -9,6 +9,52 @@ namespace ChatGptNet;
 /// </summary>
 public static class ChatGptServiceCollectionExtensions
 {
+    /// <summary>
+    /// Registers a <see cref="ChatGptClientFactory"/> instance with the specified options.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to add services to.</param>
+    /// <param name="setupAction">The <see cref="Action{ChatGptOptions}"/> to configure the provided <see cref="ChatGptOptions"/>.</param>
+    /// <returns>A reference to this instance after the operation has completed.</returns>
+    public static IServiceCollection AddChatGptClientFactory(this IServiceCollection services, Action<ChatGptOptions> setupAction)
```

Comment:
If I'm creating a factory, there is the case in which I don't want to directly provide an Action to setup `ChatGtpOptions`, so this parameter should be nullable and optional.

Reply:
Added now an `AddChatGptClientFactory(this IServiceCollection services)` that allows to customize later the client